x-pack/solutions/security/plugins/cloud_security_posture/server/routes/graph/fetch_graph.ts

```diff
@@ -44,30 +57,50 @@ export const fetchGraph = async ({
     }
   });
 
+  let isEnrichPolicyExists = false;
+  try {
+    const { policies } = await esClient.asCurrentUser.enrich.getPolicy({
+      name: GENERIC_ENTITY_INDEX_ENRICH_POLICY,
+    });
+
+    isEnrichPolicyExists = policies.some(
+      (policy) => policy.config.match?.name === GENERIC_ENTITY_INDEX_ENRICH_POLICY
+    );
+  } catch (error) {
+    logger.error(`Error fetching enrich policy ${error.message}`);
+    logger.error(error);
+    // If we can't check the policy, assume it doesn't exist and continue without enrichment
+    isEnrichPolicyExists = false;
+  }
+
   const SECURITY_ALERTS_PARTIAL_IDENTIFIER = '.alerts-security.alerts-';
   const alertsMappingsIncluded = indexPatterns.some((indexPattern) =>
     indexPattern.includes(SECURITY_ALERTS_PARTIAL_IDENTIFIER)
   );
 
-  const query = `FROM ${indexPatterns
-    .filter((indexPattern) => indexPattern.length > 0)
-    .join(',')} METADATA _id, _index
-| WHERE event.action IS NOT NULL AND actor.entity.id IS NOT NULL
-// Origin event and alerts allow us to identify the start position of graph traversal
-| EVAL isOrigin = ${
+  // Common parts of the query
+  const originEventClause =
```

Comment:
_nit_: good thinking, I would advise to split the code in a more top-bottom/tree logical way

right now you have the logic of building the ESQL query spread between two functions - `fetchGraph` and `buildEsqlQuery`. So already there's a design that is misleading. 

If I analyze `fetchGraph` logic it goes:

1. validation
   1. validate index patterns
1. query preparation phase
   1. understand whether enrich policy exists
3. build query
    1. construct each clause individually in your style
    2. group them together
5. execute query

So I'd stick with something in that form